homeassistant/components/script/__init__.py

```diff
@@ -319,6 +319,10 @@ def __init__(self, hass, object_id, cfg, raw_config, blueprint_inputs):
         self.description = cfg[CONF_DESCRIPTION]
         self.fields = cfg[CONF_FIELDS]
 
+        # The object ID of scripts need / are unique already
+        # they cannot be changed from the UI after creating
+        self._attr_unique_id = object_id
```

Comment:
hmm actually I don't think this will work.

Reply:
Consider this YAML:

```
script hello:
  example:
    ...

script hello_too:
  example:
    ...
```

Reply:
It does work, as its already bit sketchy actually 😮 

```py
async def async_validate_config(hass, config):
    """Validate config."""
    scripts = {}
    for _, p_config in config_per_platform(config, DOMAIN):
        for object_id, cfg in p_config.items():
            cfg = await _try_async_validate_config_item(hass, object_id, cfg, config)
            if cfg is not None:
                scripts[object_id] = cfg
```

Meaning if a script object ID exists twice, it gets overwritten.
Thus never the same one exists twice.

I think it would be nice to throw a warning in the config validation (instead of just silently overwriting). Will do that in a separate PR.

Reply:
I don't know how to handle this globally but the problem of something in YAML overriding something else exists not only for scripts. I notice this few times that users want to device two entities/devices so they add a section twice and complain why only the last one is working. If we have a way to globally catch this and log a warning it would be better.

Reply:
I think you are referring to duplicate keys in YAML: which is kinda similar, but not the issue discussed in this case.

Having duplicate keys in YAML will actually be logged (we even have tests for that):

![image](https://user-images.githubusercontent.com/195327/164470798-d284fe2e-fe6b-4c1b-bd9e-e9964e101afc.png)